hare/haretypes.go

```diff
@@ -0,0 +1,109 @@
+package hare
+
+import (
+	"github.com/spacemeshos/go-spacemesh/hare/pb"
+	"hash/fnv"
+)
+
+type Bytes32 [32]byte
+type Signature []byte
+
+type PubKey struct {
+	Bytes32
+}
+type BlockId struct {
+	Bytes32
+}
+type LayerId struct {
+	Bytes32
+}
+type MessageType byte
+
+const (
+	Status   MessageType = 0
+	Proposal MessageType = 1
+	Commit   MessageType = 2
+	Notify   MessageType = 3
+)
+
+type Identifiable interface {
+	Id() uint32
+}
+
+func NewBytes32(buff []byte) Bytes32 {
+	x := Bytes32{}
+	copy(x[:], buff)
+
+	return x
+}
+
+func (b32 *Bytes32) Id() uint32 {
+	h := fnv.New32()
+	h.Write(b32[:])
+	return h.Sum32()
+}
+
+func (b32 *Bytes32) Bytes() []byte {
+	return b32[:]
+}
+
+type Set struct {
+	blocks []BlockId
+}
+
+func NewSet(data [][]byte) Set {
+	s := Set{}
+
+	s.blocks = make([]BlockId, len(data))
+	for i := 0; i < len(data); i++ {
+		copy(s.blocks[i].Bytes(), data[i])
+	}
+
+	return s
+}
+
+func (s *Set) To2DSlice() [][]byte {
+	slice := make([][]byte, len(s.blocks))
+	for i := 0; i < len(s.blocks); i++ {
+		copy(slice[i], s.blocks[i].Bytes())
```

Comment:
ditto

Reply:
make(type, len, capacity). I passed len, not capacity.